src/Oscoin/Data/Core.hs

```diff
@@ -0,0 +1,502 @@
+-- | Core ledger data types.
+--
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -fno-warn-redundant-constraints #-}
+{-# OPTIONS_GHC -fno-warn-deprecations #-}
```

Comment:
Why is this?

Reply:
Because of the `notImplemented`s :unamused:

Reply:
This module does not use `notImplemented`. Can we enable the warning and the warning about redundant constraints?

Reply:
I can remove the redundant constraints suppression now though!

Reply:
Oh drat, you're right, it's only the other one which does!